src/main/java/edu/harvard/iq/dataverse/api/imports/ImportServiceBean.java

```diff
@@ -268,116 +275,132 @@ public Dataset doImportHarvestedDataset(DataverseRequest dataverseRequest, Harve
+                }
+                // And if we already have a dataset with this same id, in this same
+                // dataverse, but it is a LOCAL dataset (can happen!), we're going to 
+                // skip it also: 
+                if (!existingDataset.isHarvested()) {
+                    throw new ImportException("A LOCAL dataset with the global id " + globalIdString + " already exists in this dataverse; skipping.");
+                }
+                // For harvested datasets, there should always only be one version.
+                // We will replace the current version with the imported version.
+                // @todo or should we just destroy any extra versions quietly?
+                if (existingDataset.getVersions().size() != 1) {
+                    throw new ImportException("Error importing Harvested Dataset, existing dataset has " + existingDataset.getVersions().size() + " versions");
+                }
+                                
+                harvestedVersion = parser.parseDatasetVersion(obj.getJsonObject("datasetVersion"));
+                Dataset tempDataset = createTemporaryHarvestedDataset(harvestedVersion);
+                // Temporarily attach the temporary dataset to the parent Collection: 
+                // (this will be needed for the purposes of looking up the configured
+                // metadatablocks and such)
+                tempDataset.setOwner(owner);
+            }
+        
+            // Either a full new import, or an update of an existing harvested
+            // Dataset, perform some cleanup on the new version imported from the 
+            // parsed metadata:
+            
+            harvestedVersion.setDatasetFields(harvestedVersion.initDatasetFields());
 
-            if (ds.getVersions().get(0).getReleaseTime() == null) {
-                ds.getVersions().get(0).setReleaseTime(oaiDateStamp);
+            if (harvestedVersion.getReleaseTime() == null) {
+                harvestedVersion.setReleaseTime(oaiDateStamp);
             }
             
             // Check data against required contraints
-            List<ConstraintViolation<DatasetField>> violations = ds.getVersions().get(0).validateRequired();
+            List<ConstraintViolation<DatasetField>> violations = harvestedVersion.validateRequired();
             if (!violations.isEmpty()) {
-                // For migration and harvest, add NA for missing required values
+                // ... and fill the missing required values with "NA"s:
                 for (ConstraintViolation<DatasetField> v : violations) {
                     DatasetField f =  v.getRootBean();
                     f.setSingleValue(DatasetField.NA_VALUE);
                 }
             }
-
+            
+            // is this the right place to call tidyUpFields()? 
```

Comment:
I guess removing blank fields is useful before validating further, but do you need to DatasetVersion.validateRequired above if you're going to DatasetVersion.validate() below?

Reply:
Again, adopted this from the old code without a second thought. It's done like this, in separate passes in `doImport()` (the non-harvesting import method) as well. Maybe the 2 validate methods did not use to overlap? 
But, sure, it does seem like `validate()` should be enough.